### PR TITLE
Axel/conversions

### PR DESCRIPTION
**Pull request by Shelme74**
This new module hosts code for the `.convert` command. More info about its expected behavior and usage will be explained in the Discord server.

### CODE REVIEW

File: Floofbot/Modules/Conversions.cs
Comment (iShauny) [inline Floofbot/Modules/Conversions.cs:34]: Might be worth just compressing these lines, not sure it needs to be declared as a string first, could also just add these into an array of some sort and iterate it. Seems excessive to have so many if statements and repeating code. Either way, I think it would be best if you could compress this code a little

File: Floofbot/Modules/Conversions.cs
Comment (iShauny) [inline Floofbot/Modules/Conversions.cs:36]: Might be worth adding an else statement. One that lists the available conversions.

File: Floofbot/Modules/Conversions.cs
Comment (iShauny) [inline Floofbot/Modules/Conversions.cs:9]: Does this need to be an entire module? I think it would be better in utilities

**Review by iShauny — CHANGES_REQUESTED** [review]

File: Floofbot/Modules/Conversions.cs
Comment (Shelme74) [reply Floofbot/Modules/Conversions.cs]: I just fixed that, along with compressing the code.

File: Floofbot/Modules/Conversions.cs
Comment (Shelme74) [reply Floofbot/Modules/Conversions.cs:9]: Might be worth moving it to utilities, I coded it in a different file for clarity on my end.

**Comment by Shelme74** [comment]
Many commits have appeared here, I just rebased the branch and didn't think it would have shown up here

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:265]: Not needed, there's already a readonly static at the top

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:263]: I'm not sure this needs to be a class, can just be a command

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:287]: Variables should be likeThis not LikeThis ^^

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:284]: is there a need to use groups? I believe match returns the first result anyway, or am I missing something?

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:307]: where are these variables used? I can't see

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:316]: I dont think you need explicit casting here as they are already explicitly defined during initialisation.

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:322]: (optional) You can also use .IsMatch instead of .Success

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:337]: Actually, looking at it now, I would just move the rounding to the variable declaration

`double km = Math.Round(mi.Kilometers, 3, MidpointRounding.ToEven)`

**Review by iShauny — CHANGES_REQUESTED** [review]
I think this looks a lot better! A few picky things from me that I think would help keep formatting consistent. We wanna try to make sure the formatting through the whole program is the same! Good work though! I would like to see this reworked in a future PR though to remove the if statements for a tidier solution to allow expandability for other units

File: Floofbot/Modules/Utilities.cs
Comment (Shelme74) [reply Floofbot/Modules/Utilities.cs:284]: Actually, the match returns the whole string, that's why I needed to use the groups

File: Floofbot/Modules/Utilities.cs
Comment (Shelme74) [reply Floofbot/Modules/Utilities.cs:307]: the `g` var is used in the `Convert.ToString()` and the `celStr` is used in the next line in the `ToDouble` conversion. Then it's used to convert units.

I know, that's not optimal ><

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [reply Floofbot/Modules/Utilities.cs:284]: You can use positive lookahead instead of groups, eg

`\d+(?=f )`

File: Floofbot/Modules/Utilities.cs
Comment (Shelme74) [reply Floofbot/Modules/Utilities.cs:265]: Actually it is, because it's a class. However, related to the above comment, I'll make it a command to it won't be needed anymore.

**Comment by Shelme74** [comment]
Code has been updated to take comments into account. I think it still can be optimized, but it works fine there.

**Comment by Shelme74** [comment]
Changed have been made to improve the code, including the addition of negative temperatures support.

Ready for review

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:264]: Remember a parameter description and also use [Remainder] to take the whole rest of the command, refer to other commands in utilities for this

File: Floofbot/Modules/Utilities.cs
Comment (iShauny) [inline Floofbot/Modules/Utilities.cs:280]: [OPTIONAL]

This is good for a first iteration, however, might be good to take the embed builder separate, then append to the description, then remove the 'else' in the 'else if' so that multiple conversations can be made on the same command. eg 

At the top of the command
```
string embedDescription = "";
```

Then in each command:

```
EmbedDescription += "🌡 {fah} is equal to {cel}°C.\n";
```

then at the end after all the ifs 
```
                EmbedBuilder builder = new EmbedBuilder()
                {
                    Title = "Conversion",
                    Description = embedDescription,
                    Color = EMBED_COLOR
                };
```
I think this is a cleaner, more handy way to use this command

**Review by iShauny — CHANGES_REQUESTED** [review]
I think just do the parameter description change (refer to other commands with parameters for this) and maybe put the [Remainder] part and then it will be good to go, would be great if you could implement the other change but its optional for a first iteration

File: Floofbot/Modules/Utilities.cs
Comment (Shelme74) [reply Floofbot/Modules/Utilities.cs]: I updated the code with the new variable, however I didn't remove the `else` because the code doesn't handle more than one parameter at this time. I shall work on it later, though.

Comment (Shelme74) [reply Floofbot/Modules/Utilities.cs]: > however I didn't remove the `else` because the code doesn't handle more than one parameter at this time. I shall work on it later, though.
 Don't mind this, it works with the `[Remainder]`

Comment (Shelme74) [reply Floofbot/Modules/Utilities.cs]: It works but it keeps only the first number inputted, I'll work on it later on to see where it comes from.
Probably something to do with the Regex'ex.

**Comment by Shelme74** [comment]
Might be good to go, let me know if other changes are required.

**Comment by Shelme74** [comment]
Let me know if any changes are required so I'll work on them, or if something is missing

**Comment by Shelme74** [comment]
Well, is there any news here? PR is, I think, ready to be merged since last August, and as I don't have write access I can't do it myself...
If something still needs to be updated, I'll take time to do so.

**Comment by iShauny** [comment]
@Shelme74 resolve the merge conflict and I'll merge

**Comment by Shelme74** [comment]
Conflicts have been fixed, I took the opportunity to fix a missing bracket which was probably accidentally deleted while removing conflict markers. Also UnitsNET has been updated to its latest version.

**Comment by Llusus** [comment]
Merging.

**Pull request merged**
